adaptive/runner.py

```diff
@@ -494,6 +496,19 @@ def __init__(
             def goal(_):
                 return False
 
+        if executor is None:
+            if isinstance(learner.function, types.LambdaType):
```

Comment:
this check doesn't actually check if the function is a `lambda`. Check it out:

```
import types

def f():
    pass

print(isinstance(f, types.LambdaType))
```

prints `True`

Reply:
This is why the tests are failing AFAICT

Reply:
I've made the check a bit simpler.